outlines/fsm/fsm.py

```diff
@@ -103,13 +100,13 @@ def next_state(self, state: FSMState, token_id: int, idx: int = 0) -> FSMState:
 
         return FSMState(0)
 
-    def is_final_state(self, state: FSMState, idx: int = 0) -> bool:
+    def is_final_state(self, state: FSMState) -> bool:
         """Determine whether the current state of the FSM is a final state."""
         return state in self.final_states
 
-    def reset(self) -> None:
-        """Reset the FSM to its initial state. Here this only resets the token counter."""
-        self.num_tokens_generated = 0
+    def copy(self) -> "StopAtTokenFSM":
```

Comment:
Is the copy mechanism only necessary because we enforce `max_token` here? Same question for the other FSM. #451 is removing this mechanism from the FSMs.

Reply:
Yes, correct! If `max_tokens` is handled elsewhere, as suggested in https://github.com/outlines-dev/outlines/pull/451, then the `copy` method for both `StopAtTokenFSM` and `RegexFSM` can just be `return self`, which removes all overhead. This is a great suggestion

Reply:
Great! This comforts me in the idea that `max_tokens` should not be handled at the FSM level :)